ocaml/typing/mode_intf.mli

```diff
@@ -147,6 +143,21 @@ module type S = sig
 
     val zap_to_ceil : ('l * allowed) t -> Const.t
 
+    (* The functions to obtain the bounds are dangerous; see notes on functions
+       of same names in [solver_intf.mli] for cautions.
+
+       In addition, they are dangerous (even the non-conservative ones), because
```

Comment:
Consider the example:
```
let foo a b = ...
```
which is typed `A -> B -> C`. Let's say the mode of the whole function is `m0` and the mode of `B->C` is `m1.`.

The following two combinations are valid:
- m0 is uncontended and m1 is unsync.
- m0 is contended and m1 is sync.

Therefore, the lower bound of `m0` is `uncontended`, and lower bound of `m1` is `sync`. However, one must know that the two lower bounds can't holds simultaneously - and that's the risk the commented functions are exposing to the caller.

Reply:
Actually now I think about it, maybe it's better to expose `check_const_conservative` instead of those individual bounds. This way we avoid the risk said in the previous comment.

Reply:
That sounds like a good compromise.